### ISSUE TITLE
Querying results in infinite loop on 0.9.0

### ISSUE THREAD

**Issue by JorinTielen** [Bogdanp/anom-py#18]
This happens only on 0.9.0. When quering an entity on a production/remote datastore and not a local datastore emulator, anom gets stuck in an infinite loop. This example in a python interactive shell is enough to get stuck (provided you have authenticated with the gcloud tool):
```
import models
[x for x in models.ExampleModel.query().run()]
```
ExampleModel in this case is defined like any other anom model and there are some entities in the datastore to query.

meanwhile, using gcloud_requests it works fine:
```
from google.cloud import datastore
from gcloud_requests import DatastoreRequestsProxy
client = datastore.Client(_http=DatastoreRequestsProxy(), _use_grpc=False)
[x for x in client.query(kind='ExampleModel').fetch()]
```

I have figured out it is stuck in the `while True` loop in `query.py`. With some `print()` debugging i found that remaining is `None` and `self._complete` was set, therefore never going in the `if not entities: break` statement. I added a `if self._complete: break` for a quick fix, but i think there might be a larger problem with cursors/offset/limit.

**Comment by Bogdanp** [comment]
*sigh* how the fuck is this not a breaking change? Shows you how much the google-cloud-python developers care about their users. They changed how `next_page_token` works at some point so now it returns `None` after the last batch of results (which is stupid because now you can't get a cursor to after the last result of a query) and that change, as you've pointed out, only really shows up when you test against a production datastore since the emulator is lagging behind.

I'm going to push a fix in a few minutes. I have no idea why offsets don't work, but AFAICT anom is doing everything correctly according to the `google-cloud-datastore` docs.

EDIT: Here it is: https://googleapis.github.io/google-cloud-python/latest/datastore/changelog.html#id15

**Comment by Bogdanp** [comment]
OK, this should now be fixed in 0.9.1.
